### PR TITLE
Add "ndim" property to tensor

### PR DESCRIPTION
**Pull request by umanwizard**
For compatibility with numpy.

### CODE REVIEW

**Comment by SsnL** [comment]
Is `.T` coming soon? :)

**Comment by umanwizard** [comment]
@SsnL Yes

**Review by colesbury — COMMENTED** [review]
Needs some minimal testing, otherwise looks good.

We should also figure out a strategy for documentation for the new NumPy properties, functions, and kwargs. I think that Tensor properties (`ndim`, `.T`) are important enough to be documented directly in the Tensor documentation (tensors.rst). The kwarg aliases can probably go in a separate "NumPy" compatibility section

**Comment by fmassa** [comment]
Do we have a plan for `.size`? It exists for both numpy and PyTorch, but have different meanings.
Fortunately the return types are different between both (in PyTorch, it's a method while in numpy it's an attribute), but I'm not sure if there is a way of keeping both without some magic.

**Comment by umanwizard** [comment]
@fmassa I don't have any better answer than "we need to think more about what to do in those cases". So in the immediate future I'm not planning on changing anything related to `size`.

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -3191,3 +3191,8 @@ def callable(a, b) -> number
                r"""
 Is the :class:`torch.device` where this Tensor is.
 """)
+
+add_docstr_all('ndim',
+               r"""
+Is the number of dimensions in this Tensor.
```
Comment (gchanan) [inline torch/_tensor_docs.py]: can we just document it the same as `ndimension`?

**Review by gchanan — APPROVED** [review]

**Pull request closed without merging**